sync-core/src/main/java/com/cloudant/sync/query/QueryValidator.java

```diff
@@ -146,6 +167,117 @@
         return accumulator;
     }
 
+    @SuppressWarnings("unchecked")
+    private static List<Object> compressMultipleNotOperators(List<Object> clause) {
+        List<Object> accumulator = new ArrayList<Object>();
+
+        for (Object fieldClause: clause) {
+            Object predicate;
+            String fieldName;
+            // if this isn't a map, we don't know what to do so add the clause
+            // to the accumulator to be dealt with later as part of the final selector
+            // validation.
+            if (fieldClause instanceof Map && !((Map) fieldClause).isEmpty()) {
+                Map<String, Object> fieldClauseMap = (Map<String, Object>) fieldClause;
+                fieldName = (String) fieldClauseMap.keySet().toArray()[0];
```

Comment:
this `keySet().toArray()[0];` pattern arises in lots of places. I guess because of the basic structure of mongo queries you only ever expect there to be one key.

I know it's a bit late in the game to raise this, but it seems a bit pointless to use Map<K,V> all over the place if there is only going to be one key. Maybe on iOS it makes a bit more sense because of the Map literal support. But really these instances are Pair<String,Object> (2-tuple or whatever you want to call it)